packages/core/strapi/lib/commands/actions/build-command/action.js

```diff
@@ -1,18 +1,47 @@
 'use strict';
 
-const strapi = require('../../..');
+const tsUtils = require('@strapi/typescript-utils');
+
 const { buildAdmin } = require('../../builders');
 
+/**
+ * @typedef BuildActionArgs
+ * @property {"vite" | "webpack"} builder
+ */
+
 /**
  * `$ strapi build`
+ *
+ * @type {({ builder }: BuildActionArgs) => Promise<void>}
  */
-module.exports = async ({ optimization, forceBuild = true }) => {
-  const { appDir, distDir } = await strapi.compile();
+module.exports = async ({ builder }) => {
+  const cwd = process.cwd();
+
+  const isTSProject = await tsUtils.isUsingTypeScript(cwd);
+
+  let distDir = cwd;
+
+  /**
+   * @type {string | undefined}
+   */
+  let tsconfig;
+
+  if (isTSProject) {
+    distDir = await tsUtils.resolveOutDir(cwd);
+    tsconfig = await tsUtils.getConfigPath(cwd);
+  }
+
+  /**
+   * @note
+   *
+   * Currently this just builds the admin – should this actually build the plugins in your application too?
```

Comment:
Do you mean src/plugins plugins or external plugins ?

Reply:
I mean a strapi user's plugins within their local application – it might not be necessary :)

Reply:
Yes ok, considering what we disucssed internaly we probably want this too be supported too. We can add this when we start putting those tasks in the sprints :)